config.m4

```diff
@@ -206,7 +206,6 @@ if test "$MONGODB" != "no"; then
     bson-atomic.c \
     bson-clock.c \
     bson-context.c \
-    bson-decimal128.c \
```

Comment:
Weren't we going to leave it compiled in? AFAIK, @bjori said there was a feature flag protection in it.